### PR TITLE
Morecmek

### PR DESCRIPTION
**Pull request by RemoveRusky**

### CODE REVIEW

File: CompilerSource/languages/language_adapter.h [hunk 1]
```diff
@@ -35,12 +35,12 @@ struct language_adapter {
...
   // Sizable utilities
-  virtual int link_globals(parsed_object*, EnigmaStruct*, parsed_script*[], vector<parsed_script*>& tlines) = 0;
-  virtual int link_ambiguous(parsed_object*, EnigmaStruct*, parsed_script*[], vector<parsed_script*>& tlines) = 0;
+  virtual int link_globals(parsed_object*, EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines) = 0;
+  virtual int link_ambiguous(parsed_object*, EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines) = 0;
 
   // IDE_EDITABLEs added before compile
-  virtual int compile_parseAndLink(EnigmaStruct*,parsed_script*[], vector<parsed_script*>& tlines, const std::set<std::string>& script_names) = 0;
-  virtual int compile_parseSecondary(map<int,parsed_object*>&,parsed_script*[],int scrcount, vector<parsed_script*>& tlines, map<int,parsed_room*>&,parsed_object*, const std::set<std::string>&) = 0;
+  virtual int compile_parseAndLink(EnigmaStruct*, vector<parsed_script*>&, vector<parsed_script*>& tlines, const std::set<std::string>& script_names) = 0;
```
Comment (JoshDreamland) [inline CompilerSource/languages/language_adapter.h]: Prefer `const&` so it's clear nothing is being mutated (the stuff behind the pointers may be, but that's fine).

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  CMAKE_ARGS ${CL_ARGS}
+)
+
+ExternalProject_Add(
+  compiler
+  DOWNLOAD_COMMAND ""
+  SOURCE_DIR "${CMAKE_SOURCE_DIR}/CompilerSource"
+  BINARY_DIR "${CMAKE_SOURCE_DIR}/build/compiler"
+  CMAKE_ARGS ${CL_ARGS}
+)
```
Comment (RobertBColton) [inline CMakeLists.txt]: You're missing a newline here.

Comment (fundies) [reply CMakeLists.txt]: this pr should be closed. idk y some idiot reopened it

**Comment by RobertBColton** [comment]
Closing in favor of #1642 which was recreated from this. The branch can be deleted some period of time after this has been closed.

**Pull request closed without merging**